meerkat_frontend/templates/explore/index.html

```diff
@@ -388,37 +387,56 @@
 
      reDraw();
 
-}
-
-
-// Variables to keep track of if we have drawn a timeline and with what id and category
-var timelineDrawn = false;
-var timelineId = "";
-var timelineCategory = "";
-
-// Find category and draw timeline
-function prepareExploreTimeline(id, axis){
-    options = {};
-    $('.timeline-option').each( function(){
-        options[$(this).attr('name')] = $(this).attr('value');
-    });
-    console.log(options)
-    if( axis == "x"){
-        category = $('.yaxis .axis-selector select').val();
-    } else {
-        category = $('.xaxis .axis-selector select').val();
-    }
-    createTimeline(
-        id,
-        category,
-        options
-    );
-    timelineDrawn= true;
-    timelineId = id;
-    timelineCategory = category;
-    $('#timeline').show();
-    // Scrolls down to the timeline
-    $('html, body').animate({ scrollTop: $('#timeline-wrapper').offset().top }, 'slow');
+ }
+
+
+ // Variables to keep track of if we have drawn a timeline and with what id and category
+ var timelineDrawn = false;
+ var timelineId = "";
+ var timelineCategory = "";
+ var timelineOptions = {};
+
+ // Find category and draw timeline
+ function prepareExploreTimeline(id, axis){
+
+     options = {};
+     $('.timeline-option').each( function(){
+         options[$(this).attr('name')] = $(this).attr('value');
+     });
+     var sec_category;
+
+     if( axis == "x"){
+         category = $('.yaxis .axis-selector select').val();
+         sec_category = $('.xaxis .axis-selector select').val();
+     } else {
+         category = $('.xaxis .axis-selector select').val();
+         sec_category = $('.yaxis .axis-selector select').val();
+     }
+
+     //API call have to be slightly different when "location" is a variable to query vs. a category.
+     options.location_case = false;
+     if(( sec_category == "locations:district" ) ||
+        ( sec_category == "locations:region" ) ||
+        ( sec_category == "locations:clinic" )
+     ){
+         options.location_case = true;
+         options.location = id;
+     }
+
+     createTimeline(
+         id,
+         category,
+         options
+     );
+
+     timelineDrawn= true;
+     timelineId = id;
+     timelineCategory = category;
+     timelineOptions = options;
+
+     $('#timeline').show();
+     // Scrolls down to the timeline
+     $('html, body').animate({ scrollTop: $('#timeline-wrapper').offset().top }, 'slow');
```

Comment:
Lines 393-439 a really nice neat piece of code. :+1:

Reply:
it is not mine :P I think it is yours!